### PR TITLE
Basic metric agent setup

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Deploy an OTEL collector as an agent (daemonset) if the input is defined in any pipeline. The most basic configuration is used
- Refactor resource management (gateway vs agent mode)
- Refactor config building. Refine the `config/builde/otlpoutput` API
- Introduce test data builder for reconciler unit tests

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/17373

### CODE REVIEW

**Review by NHingerl — APPROVED** [review]

File: internal/resources/otelcollector/core/core.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+func MakeDefaultLabels(baseName string) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name": baseName,
+	}
+}
+
+func MakePodSpec(baseName, image, priorityClassName string, resources corev1.ResourceRequirements) corev1.PodSpec {
+	labels := MakeDefaultLabels(baseName)
+	affinity := makePodAffinity(labels)
```
Comment (chrkl) [inline internal/resources/otelcollector/core/core.go:29]: Not required for the DaemonSet. But probably would also not cause any harm to keep it. Please check again.

File: main.go [hunk 2]
```diff
@@ -515,56 +523,67 @@ func createLogParserValidator(client client.Client) *logparserwebhook.Validating
...
-			BaseCPURequest:       resource.MustParse(metricGatewayCPURequest),
-			DynamicCPURequest:    resource.MustParse(metricGatewayDynamicCPURequest),
-			BaseMemoryRequest:    resource.MustParse(metricGatewayMemoryRequest),
-			DynamicMemoryRequest: resource.MustParse(metricGatewayDynamicMemoryRequest),
+	config := metricpipeline.Config{
+		Agent: agent.Config{
+			Namespace: telemetryNamespace,
+			BaseName:  "telemetry-metric-agent",
+			DaemonSet: agent.DaemonSetConfig{
+				Image: traceCollectorImage,
```
Comment (chrkl) [inline main.go:563]: Set same priority class as for Fluent Bit

File: main.go [hunk 3]
```diff
@@ -515,56 +523,67 @@ func createLogParserValidator(client client.Client) *logparserwebhook.Validating
...
-			DynamicMemoryLimit:   resource.MustParse(metricGatewayDynamicMemoryLimit),
-			BaseCPURequest:       resource.MustParse(metricGatewayCPURequest),
-			DynamicCPURequest:    resource.MustParse(metricGatewayDynamicCPURequest),
-			BaseMemoryRequest:    resource.MustParse(metricGatewayMemoryRequest),
-			DynamicMemoryRequest: resource.MustParse(metricGatewayDynamicMemoryRequest),
+	config := metricpipeline.Config{
+		Agent: agent.Config{
+			Namespace: telemetryNamespace,
+			BaseName:  "telemetry-metric-agent",
+			DaemonSet: agent.DaemonSetConfig{
```
Comment (chrkl) [inline main.go:562]: Maybe also define the resources here to have that in a unified way

**Review by NHingerl — DISMISSED** [review]

File: main.go (hunk 2 above)
Comment (skhalash) [reply main.go]: Done

File: apis/telemetry/v1alpha1/metricpipeline_types.go [hunk 4]
```diff
@@ -20,19 +20,74 @@ import (
...
+// MetricPipeline is the Schema for the metricpipelines API.
+type MetricPipeline struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MetricPipelineSpec   `json:"spec,omitempty"`
+	Status MetricPipelineStatus `json:"status,omitempty"`
+}
+
+// MetricPipelineSpec defines the desired state of MetricPipeline.
```
[the remark this answers is not in the document]
Comment (skhalash) [reply apis/telemetry/v1alpha1/metricpipeline_types.go:55]: Improved the docs

**Review by NHingerl — APPROVED** [review]

**Review by chrkl — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**